### PR TITLE
DOC: Improve sort parameter doc of DataFrameGroupBy.value_counts

### LINKED ISSUE #59307: BUG: GroupBy.value_counts doesn't preserve original order for non-grouping rows
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
df = pd.DataFrame({"X": ["B", "A", "A", "B"], "Y": [4, 1, 3, -1]})
df.groupby("X", sort=False).value_counts(sort=False)  # works as expected
df.groupby("X", sort=True).value_counts(sort=False)  # column "Y" does not match original order
```

### Issue Description

(possibly related to #55951)

When a GroupBy.value_counts operation is performed, the order of rows within the non-grouping columns does not respect the order of elements in the original frame.

In this example, with value_counts(sort=False) I would expect the row `B 4 1` to appear above `B -1 1`, as that would correspond go their order in the original frame.
```
>>> df = pd.DataFrame({"X": ["B", "A", "A", "B"], "Y": [4, 1, 3, -1]})
>>> df.groupby("X", sort=False).value_counts(sort=False)
X  Y 
A   1    1
    3    1
B  -1    1
    4    1
Name: count, dtype: int64
```

When the groupby has sort=False set, this order is respected.
```
>>> df.groupby("X", sort=False).value_counts(sort=False)
X  Y 
B   4    1
A   1    1
    3    1
B  -1    1
Name: count, dtype: int64
```

### Expected Behavior

Calling groupby(sort=True).value_counts(sort=False) should preserve the order of members within groups, or documentation should be changed to reflect this.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : 67a58cddc2f8c0e30cb0123589947d9b3f073720
python                : 3.11.7
python-bits           : 64
OS                    : Darwin
OS-release            : 23.5.0
Version               : Darwin Kernel Version 23.5.0: Wed May  1 20:14:38 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6020
machine               : arm64
processor             : arm
byteorder             : little
LC_ALL                : None
LANG                  : en_US.UTF-8
LOCALE                : en_US.UTF-8

pandas                : 3.0.0.dev0+1239.g67a58cddc2
numpy                 : 2.1.0.dev0+git20240720.d489c83
pytz                  : 2024.1
dateutil              : 2.9.0.post0
pip                   : 23.3.1
Cython                : None
sphinx                : None
IPython               : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : None
blosc                 : None
bottleneck            : None
fastparquet           : None
fsspec                : None
html5lib              : None
hypothesis            : None
gcsfs                 : None
jinja2                : None
lxml.etree            : None
matplotlib            : None
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
psycopg2              : None
pymysql               : None
pyarrow               : None
pyreadstat            : None
pytest                : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : None
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
xlsxwriter            : None
zstandard             : None
tzdata                : 2023.4
qtpy                  : None
pyqt5                 : None
</details>

### PR DESCRIPTION
**Pull request by KevsterAmp**
- [x] closes #59307 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

I just copied the sort docstring of `Dataframe.value_counts` since the latter is mirrored to behaved like it.

image

### CODE REVIEW

**Comment by KevsterAmp** [comment]
Seems like I have changed the wrong `value_count()` function. I have changed `pandas.core.groupby.SeriesGroupBy.value_counts` rather than `pandas.core.groupby.DataFrameGroupBy.value_counts`. Applying fix in a bit

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -2303,7 +2303,7 @@ def value_counts(
         normalize : bool, default False
             Return proportions rather than frequencies.
         sort : bool, default True
-            Sort by frequencies.
+            Sort by frequencies when True. Sort by DataFrame column values when False.
```
Comment (rhshadrach) [inline pandas/core/groupby/generic.py:2306]: I think we also want to modify the docstring on `SeriesGroupBy.value_counts`. Can you make the same adjustment there.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (KevsterAmp) [reply pandas/core/groupby/generic.py:2306]: Sure thing

**Comment by KevsterAmp** [comment]
@rhshadrach - are there any extra changes needed for this PR? Thanks

**Comment by sfc-gh-joshi** [comment]
@KevsterAmp I left an additional ? on the parent issue: https://github.com/pandas-dev/pandas/issues/59307#issuecomment-2271819631

It seems like when the groupby's sort flag and value_counts sort flag are both false, no sorting is done. Is this a behavioral bug, or just something else to add to documentation?

**Comment by KevsterAmp** [comment]
Fixed the doc, thanks. 

image

**Comment by KevsterAmp** [comment]
> @KevsterAmp I left an additional ? on the parent issue: [#59307 (comment)](https://github.com/pandas-dev/pandas/issues/59307#issuecomment-2271819631)
> 
> It seems like when the groupby's sort flag and value_counts sort flag are both false, no sorting is done. Is this a behavioral bug, or just something else to add to documentation?
> 
> **edit:** It seems also that when `value_counts` has sort=True, the result frame is still being sorted on the DataFrame column values:
> 
> ```
> >>> data0 = {
>     "by": ["c", "b", "a", "a", "b", "b", "c", "a"],
>     "value1": ["ee", "aa", "bb", "aa", "bb", "cc", "dd", "aa"],
>     "value2": [1, 2, 3, 1, 1, 3, 2, 1],
> }
> >>> pd.DataFrame(data0).groupby(by=["by"]).value_counts(sort=True)
> by  value1  value2
> a   aa      1         2
>     bb      3         1
> b   aa      2         1
>     bb      1         1
>     cc      3         1
> c   dd      2         1
>     ee      1         1
> ```
> 
> In the input frame, the row `c ee 1` (which appears once) appears before `c dd 1` (also appearing once), but the result ends up being sorted on `value1` as well as the counts. I think it's worth clarifying whether `value_counts(sort=True)` will also sort on the DataFrame's values, or if it's just that the algorithm used to sort the counts column is unstable.

Not sure if I will address this in this PR, since the scope of this one is to just modify the doc. Thanks

**Comment by KevsterAmp** [comment]
@rhshadrach - are there any extra changes needed for this PR? Thanks

**Comment by rhshadrach** [comment]
@KevsterAmp - I am hesitant to make this change since certain cases that are correct now become incorrect. When you do `df.groupby(..., sort=False).value_counts(sort=False)`, pandas is preserving the order of the input. See https://github.com/pandas-dev/pandas/issues/59307#issuecomment-2271819631

**Comment by rhshadrach** [comment]
@KevsterAmp - I've opened #59307 based on the discussion in the linked issue. Unfortunately that supersedes this PR. Going to close here - but thanks for working on this!

**Pull request closed without merging**